src/common/io_exerciser/DataGenerator.cc

```diff
@@ -0,0 +1,713 @@
+{
+  try
+  {
+    UniqueIdBytes read_unique_run_id = readUniqueRunId(block_offset - read_offset,
+                                                       bufferlist);
+    if (DataGenerationSingleton::getUniqueId() != read_unique_run_id)
+    {
+      return ErrorType::RUN_ID_MISMATCH;
+    }
+
+    SeedBytes read_seed = readSeed(block_offset - read_offset, bufferlist);
+    if (m_model.get_seed(block_offset) != read_seed)
+    {
+      return ErrorType::SEED_MISMATCH;
+    }
+
+    if (std::strncmp(&bufferlist[((block_offset - read_offset) *
+                      m_model.get_block_size()) + bodyStart()],
+                     generate_block(block_offset).c_str() + bodyStart(),
+                     m_model.get_block_size() - bodyStart()) != 0)
+    {
+      return ErrorType::DATA_MISMATCH;
+    }
+  }
+  catch(const std::exception& e)
+  {
+    return ErrorType::DATA_NOT_FOUND;
+  }
+
+  return ErrorType::UNKNOWN;
+}
+
+void HeaderedSeededRandomGenerator
+  ::printDebugInformationForBlock(uint64_t read_offset, uint64_t block_offset,
+                                  const bufferlist& bufferlist)
+{
+  ErrorType blockError = getErrorTypeForBlock(read_offset, block_offset, bufferlist);
+
+  TimeBytes read_time = 0;
+  std::time_t ttp;
+
+  char read_bytes[m_model.get_block_size()];
+  char generated_bytes[m_model.get_block_size()];
+
+  if (blockError == ErrorType::DATA_MISMATCH || blockError == ErrorType::UNKNOWN)
+  {
+    read_time = readDateTime(block_offset, bufferlist);
+    std::chrono::system_clock::time_point time_point{std::chrono::milliseconds{read_time}};
+    ttp = std::chrono::system_clock::to_time_t(time_point);
+
+    std::memcpy(&read_bytes,
+                &bufferlist[((block_offset - read_offset) * m_model.get_block_size())],
+                m_model.get_block_size() - bodyStart());
+    std::memcpy(&generated_bytes,
+                generate_block(block_offset).c_str(),
+                m_model.get_block_size() - bodyStart());
+  }
+  std::stringstream ss;
+  switch(blockError)
+  {
+    case ErrorType::RUN_ID_MISMATCH:
+    {
+      UniqueIdBytes read_unique_run_id = readUniqueRunId((block_offset - read_offset),
+                                                          bufferlist);
+      ss << "Header (Run ID) mismatch detected at block " << block_offset 
+         << " (byte offset " << block_offset * m_model.get_block_size() << ")."
+         << " Header expected run id " << DataGenerationSingleton::getUniqueId()
+         << " but found id " << read_unique_run_id
+         << ". Block data corrupt or not written from this instance of this application.";
+    }
+    break;
+    
+    case ErrorType::SEED_MISMATCH:
+    {
+      SeedBytes read_seed = readSeed((block_offset - read_offset), bufferlist);
+
+      if (m_model.get_seed_offsets(read_seed).size() == 0)
+      {
+        ss << "Data (Seed) mismatch detected at block " << block_offset
```

Comment:
I'd recommend using std::format() for all new code, instead of chains of streams

Reply:
Have updated to make use of the fmt::format library you mentioned later on rather than stringstreams.